packages/lab/src/calendar/CalendarNavigation.tsx

```diff
@@ -171,55 +175,62 @@ function useCalendarNavigation() {
       }
     },
     [
+      dateAdapter,
       isOutsideAllowedYears,
       isOutsideAllowedMonths,
       setVisibleMonth,
       visibleMonth,
-      locale,
     ],
   );
 
-  const months: DateValue[] = useMemo(
-    () => monthsForLocale(visibleMonth, locale),
-    [visibleMonth, locale],
+  const months: TDate[] = useMemo(
+    () => generateMonthsForYear<TDate>(dateAdapter, visibleMonth),
+    [visibleMonth],
   );
 
-  const years: DateValue[] = useMemo(
+  const years: TDate[] = useMemo(
     () =>
-      generateYearsBetweenRange(
-        Math.min(minDate ? minDate.year : CALENDAR_MIN_YEAR, visibleMonth.year),
-        Math.max(maxDate ? maxDate.year : CALENDAR_MAX_YEAR, visibleMonth.year),
+      generateYearsBetweenRange<TDate>(
```

Comment:
Confirming this fixes year range not respecting min/max years (SO 93573)

Reply:
yes, min/max date affects the values populated in years.
There is a edge-case where your visible year is currently 2025, and you set the min/max to 2023, the years will currently show 2025, as changing the min/max does not affect the current visible month. However, once you move back to a year within the range, the years dropdown removes 2025 as a possible year. Will follow up on this, as this PR, is know in a known state. @ivan-calderon have a play with this edge case